code/defines/obj/weapon.dm

```diff
@@ -482,27 +588,37 @@
 				var/log_str = "[key_name(usr)]<A HREF='?_src_=holder;adminmoreinfo=\ref[AM]'>?</A> has triggered an IED-rigged [name] at <A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[bombturf.x];Y=[bombturf.y];Z=[bombturf.z]'>[A.name] (JMP)</a>."
 				message_admins(log_str)
 				log_game(log_str)
+
 				spawn(IED.det_time)
 					IED.prime()
 
 					src.desc = initial(src.desc)
 
-			if(ishuman(L))
-				var/mob/living/carbon/H = AM
-				if(H.m_intent == "run")
-					armed = 0
-					H.legcuffed = src
-					src.forceMove(H)
-					H.update_inv_legcuffed()
+			else if(H.m_intent == "run") //This is where the real fun begins
+				trapped = 1
+				trappeduser = H
+				armed = 0
+
+				playsound(src, 'sound/effects/snap.ogg', 60, 1)
+				H.audible_scream()
+				lock_atom(H, /datum/locking_category/beartrap)
+
+				var/datum/organ/external/affecting = H.pick_usable_organ(LIMB_LEFT_LEG, LIMB_RIGHT_LEG)
+				if(affecting)
```

Comment:
instead of pasting this codeblock in multiple places, why not just have one proc that handles fucking up a bodypart? you can use it repeatedly for the two walk over parts and the arm as well

Reply:
I'll add it, makes sense. Thanks!